lib/plugins/aws/provider.js

```diff
@@ -1894,11 +1895,15 @@ Object.defineProperties(
       { promise: true }
     ),
     resolveImageUriAndShaFromPath: d(
-      async function (imageName, imagePath) {
+      async function (imageName, imagePath, imageFilename = 'Dockerfile') {
```

Comment:
Optional arguments are not taken into account in memoization, and I believe in that case this one should be considered.

To fix it we may either:
- Not make this argument optional (that's simplest), and always pass it to this function
- Make it optional, then we need to enforce `length: 3` setting, and ideally configure a _resolver_ for third argument, to ensure that not passing it is equal to passing `'Dockerfile'` (that can be done with option: `resolvers: [null, null, (imageFilename = 'Dockerfile') => imageFilename]`)

Reply:
Great catch, I was not aware of that :+1: I've changed the logic to always pass `imageFilename` here so memoization should work correctly :+1: